.travis.yml

```diff
@@ -57,6 +61,15 @@ jobs:
         - unbuffer node build-system/pr-check/checks.js
     - stage: build
       name: 'Validator Tests'
+      # Install bazel (not available from apt-get).
+      before_install:
+        # Override Xenial's default Java version (github.com/travis-ci/travis-ci/issues/10290)
+        - export PATH=$(echo "$PATH" | sed -e 's/:\/usr\/local\/lib\/jvm\/openjdk11\/bin//')
+        - export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64
+        - pip install --user protobuf
+        - wget https://github.com/bazelbuild/bazel/releases/download/2.2.0/bazel_2.2.0-linux-x86_64.deb
```

Comment:
We can prevent this file from being download during every single build by doing the following:
- Change the download location to a new local directory
- Add the directory to the Travis cache
- Check the cache for the file and only download if it isn't present

https://github.com/ampproject/amphtml/blob/2a86dead8cfb7e9107321de21d4f51eda243784a/.travis.yml#L111-L119

In addition, I think it's worth creating a `.sh` file with all this logic, and calling it from within `gulp validator-java`, similar to how the sauce proxy is downloaded and started before tests.

https://github.com/ampproject/amphtml/blob/2a86dead8cfb7e9107321de21d4f51eda243784a/build-system/sauce_connect/start_sauce_connect.sh#L76-L86

Reply:
All this is now done, but my comfort with Bash is low and I anticipate needing tweaks to logging, script placement, and general style. That said, I think the main pieces are all in place?

Reply:
They are indeed.